testbench_gil.py

```diff
@@ -0,0 +1,118 @@
+"""Collection of benchmarking routines for tasks related testing
+minibatch generation with Pescador.
+
+Sample Calls
+------------
+To run the tests with a bit of verbosity...
+
+  $ py.test -vs testbench_gil.py \
+    --benchmark-min-rounds=100 \
+    --benchmark-sort=mean
+
+Optionally, you can direct the test to preserve the temporary data generated
+during the tests...
+
+  $ py.test -vs --no-clean testbench_performance.py
+
+
+You need to run it with the --benchmark-save=foobar parameter
+for to successfully generate a json output file.
+
+  $ py.test -vs testbench_performance.py --benchmark-save=bench1
+"""
+import biggie
+import h5py
+import logging
+import numpy as np
+import pescador
+import pytest
+
+import theano
+import theano.tensor as T
+
+import minibench
+
+logging.basicConfig(level=logging.INFO)
+
+
+def theano_test_fx(n_dots):
+    # input vector.
+    x = T.matrix('x')
+    # A fake weight matrix.
+    w = T.matrix('w')
```

Comment:
for "true" comparison with how theano gets used, this should probably be a `shared` variable